src/main/java/org/myrobotlab/service/Security.java

```diff
@@ -429,119 +335,145 @@ public void createDefaultGroups() {
-      error("user %s does not exist can not change groupId", userId);
-      return false;
-    }
+      new File(keyFile.getParent()).mkdirs();
 
-    if (!groups.containsKey(groupId)) {
-      error("group %s does not exist can not change groupId", groupId);
-      return false;
+      KeyGenerator keyGen = KeyGenerator.getInstance(Security.AES);
+      keyGen.init(128);
+      SecretKey sk = keyGen.generateKey();
+      FileWriter fw = new FileWriter(keyFile);
+      fw.write(byteArrayToHexString(sk.getEncoded()));
+      fw.flush();
+      fw.close();
     }
 
-    users.get(userId).groupId = groupId;
-
-    return false;
-
+    SecretKeySpec sks = getSecretKeySpec(keyFile);
+    Cipher cipher = Cipher.getInstance(Security.AES);
+    cipher.init(Cipher.ENCRYPT_MODE, sks, cipher.getParameters());
+    byte[] encrypted = cipher.doFinal(passphrase.getBytes());
+    return byteArrayToHexString(encrypted);
   }
 
-  public boolean setGroup(String userId, String groupId) {
-    if (!users.containsKey(userId)) {
-      error("user %s does not exist", userId);
-      return false;
-    }
-    if (!groups.containsKey(groupId)) {
-      error("group %s does not exist", groupId);
-      return false;
+  /**
+   * get a key/secret from the secure store
+   * 
+   * @param name
+   *          - the name of the security key
+   * @return
+   */
+  public String getKey(String name) {
+    if (store.containsKey(name)) {
+      return store.getProperty(name);
     }
 
-    User u = users.get(userId);
-    u.groupId = groupId;
-    return true;
+    log.error("could not find {} in security store", name);
+    return null;
+  }
+
+  public String getKeyFileName() {
+    return keyFileName;
   }
 
   /**
-   * This method returns all the details of the class without it having to be
-   * constructed. It has description, categories, dependencies, and peer
-   * definitions.
-   * 
-   * @return ServiceType - returns all the data
-   * 
+   * return the set of key names currently stored in the key store
+   * @return
    */
-  static public ServiceType getMetaData() {
+  public Set<String> getKeyNames() {
+    Set<String> ret = new TreeSet<String>();
+    for (Object o : store.keySet()) {
+      ret.add(o.toString());
+    }
+    return ret;
+  }
 
-    ServiceType meta = new ServiceType(Security.class.getCanonicalName());
-    meta.addDescription("provides security");
-    meta.addCategory("framework", "security");
-    meta.includeServiceInOneJar(true);
+  @Deprecated // use getKey
+  public String getSecret(String name) {
+    return getKey(name);
+  }
 
-    return meta;
+  private SecretKeySpec getSecretKeySpec(File keyFile) throws NoSuchAlgorithmException, IOException {
+    byte[] key = readKeyFile(keyFile);
+    SecretKeySpec sks = new SecretKeySpec(key, Security.AES);
+    return sks;
+  }
+
+  public String getStoreFileName() {
+    return storeFileName;
+  }
+
+  private byte[] hexStringToByteArray(String s) {
+    byte[] b = new byte[s.length() / 2];

```

Comment:
same.. consider apache commons codec here instaed...

Reply:
its in Security which is core ... same response applies